src/main/java/com/mongodb/spark/sql/connector/schema/RowToBsonDocumentConverter.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.mongodb.spark.sql.connector.schema;
+
+import static java.lang.String.format;
+
+import java.sql.Date;
+import java.sql.Timestamp;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.types.ArrayType;
+import org.apache.spark.sql.types.DataType;
+import org.apache.spark.sql.types.DataTypes;
+import org.apache.spark.sql.types.Decimal;
+import org.apache.spark.sql.types.DecimalType;
+import org.apache.spark.sql.types.MapType;
+import org.apache.spark.sql.types.StringType;
+import org.apache.spark.sql.types.StructField;
+import org.apache.spark.sql.types.StructType;
+
+import org.bson.BsonArray;
+import org.bson.BsonBinary;
+import org.bson.BsonBoolean;
+import org.bson.BsonDateTime;
+import org.bson.BsonDecimal128;
+import org.bson.BsonDocument;
+import org.bson.BsonDouble;
+import org.bson.BsonInt32;
+import org.bson.BsonInt64;
+import org.bson.BsonString;
+import org.bson.BsonValue;
+import org.bson.types.Decimal128;
+
+import com.mongodb.spark.sql.connector.exceptions.DataException;
+
+interface RowToBsonDocumentConverter {
```

Comment:
I suggest changing this interface to a class for mostly the same reasons as in this [comment](https://github.com/mongodb/mongo-spark/pull/54/files#r716296848).

Reply:
Done

Reply:
I do not see related changes in the second commit, maybe they were not pushed.

But the above observation is good:), because I also noticed that `BsonDocumentToRowConverter` is an instantiable class, while `RowToBsonDocumentConverter` is a collection of static methods. I understand that `RowToBsonDocumentConverter` does not need to be instantiable, but the inconsistency in what a converter is (an object or a `static` method) is not good. I would like to change the [previous proposal](https://github.com/mongodb/mongo-spark/pull/54#discussion_r716311756) and propose to consider changing this interface not to a class with `static` methods, but to an instantiable class for consistency with `BsonDocumentToRowConverter`.

Reply:
A happy accident! I've made both converters final classes. Thanks @stIncMale